opencti-platform/opencti-front/src/private/components/arsenal/channels/Channel.jsx

```diff
@@ -58,9 +55,6 @@ class ChannelComponent extends Component {
           stixCoreObjectOrStixCoreRelationshipId={channel.id}
           defaultMarkings={channel.objectMarking ?? []}
         />
```

Comment:
```suggestion
        />
        {
          !isFABReplaced && (
            <Security needs={[KNOWLEDGE_KNUPDATE]}>
              <ChannelEdition channelId={channel.id} />
            </Security>
          )
        }
```

However this requires refactoring the component to use hooks (not possible in these old class components).
This will be the case for several other components.

Luckily I've already done most of this refactoring [in this PR ](https://github.com/OpenCTI-Platform/opencti/pull/8010). I had the same issue, using hooks.

I hope to merge in the next days.

Reply:
@labo-flg if your PR is being merged soon, I'll wait that so I can use the necessary hooks to reimplement this logic. Thanks for reviewing these!

Reply:
I've just merged https://github.com/OpenCTI-Platform/opencti/pull/8010
If you rebase you'll see that Channel.jsx has been moved to Channel.tsx, using now functional component.

If one of the files you need to change have not been refactored, let me know.